src/main/java/model/students/StudentsList.java

```diff
@@ -28,31 +39,72 @@
         return studentsMapFiltered;
     }
 
-    private StudentsList init() {
-        String[] firstNames = {"Olivia", "Amelia", "Isla", "Emily", "Ava", "Oliver", "Harry", "Jack", "George", "Noah"};
-        String[] lastNames = {"Smith", "Johnson", "Williams", "Jones", "Brown", "Davis", "Miller", "Wilson", "Moore", "Taylor"};
-        int[] ages = {18, 20, 21, 22, 18, 18, 19, 23, 23, 20};
-        int[] ids = {96, 97, 98, 99, 100, 101, 102, 103, 104, 105};
+    private void initFromFile() {
+        String fileData = readFromFile();
+
+        String[] lines = fileData.split(System.getProperty("line.separator"));
+        for (int i = 1; i < lines.length; i++) {
+            String[] data = lines[i].split(",");
+            String firstName = data[1];
+            String lastName = data[2];
+            String email = data[4];
+            String phone = data[5];
+
+            try {
+                checkContactData(email, phone);
```

Comment:
there is not needed double check of contact data. With method checkContactData and in methods withEmail and withPhone. Select one straightforward approach to do this.